src/rsa/signing.rs

```diff
@@ -125,11 +125,34 @@ impl RSAKeyPair {
                 if p.bit_length() != q.bit_length() {
                     return Err(error::Unspecified);
                 }
+                let q = try!(q.into_odd_positive());
+                let n = try!(n.into_modulus::<N>());
+
+                let q_mod_n_decoded = {
+                    let q = try!(q.try_clone());
+                    try!(q.into_elem_decoded(&n))
+                };
+
                 // XXX: |p < q| is actual OK, it seems, but our implementation
                 // of CRT-based moduluar exponentiation used requires that
                 // |q > p|. (|p == q| is just wrong.)
-                let q = try!(q.into_odd_positive());
                 try!(bigint::verify_less_than(&q, &p));
+                {
+                    let p_mod_n = {
+                        let p = try!(p.try_clone());
+                        try!(p.into_elem_decoded(&n))
+                    };
+                    let p_minus_q = {
```

Comment:
Please disregard the previous comment, since I'm not merging this yet, as I found a minor issue with the first commit.